reference-client/src/test/java/com/transformuk/hee/tis/reference/client/impl/ReferenceServiceImplTest.java

```diff
@@ -1314,6 +1319,60 @@ public void shouldCheckCurrentRolesExistsWhenCurrentOnly() {
         HttpMethod.POST, requestEntity, responseType);
   }
 
+  @Test
+  public void shouldCheckAnyRolesMatchesWhenNotCurrentOnly() {
+    // Given.
+    List<String> codes = Arrays.asList("code1", "code2");
+    HttpEntity<List<String>> requestEntity = new HttpEntity<>(codes);
+    ParameterizedTypeReference<Map<String, String>> responseType = getMatchesStringReference();
+
+    Map<String, String> response = new HashMap<>();
+    response.put("code1", "code1");
+    response.put("code2", "");
+
+    given(referenceRestTemplate
+        .exchange(anyString(), any(HttpMethod.class), any(RequestEntity.class),
+            Matchers.<ParameterizedTypeReference<Map<String, String>>>any()))
+        .willReturn(ResponseEntity.ok(response));
+
+    // When.
+    Map<String, String> matches = referenceServiceImpl.rolesMatch(codes, false);
+
+    // Then.
+    assertThat("Unexpected 'exists' result value.", matches.get("code1"), is("code1"));
```

Comment:
Last thing I think (sorry Marcello!) - 'exists' should be 'matches' here and in the test below.

Reply:
ahah it's fine, there were loads of 'exists' to weed out and swap with 'matches', I knew I must have left some behind. Well spotted!